.pipelines/templates/e2e-upgrade-template.yml

```diff
@@ -0,0 +1,97 @@
+jobs:
+  - job: e2e_upgrade_tests
+    timeoutInMinutes: 10
+    cancelTimeoutInMinutes: 5
+    workspace:
+      clean: all
+    variables:
+      - name: REGISTRY_NAME
+        value: kind-registry
+      - name: REGISTRY_PORT
+        value: 5000
+      - name: KUBERNETES_VERSION
+        value: v1.21.1
+      - name: KIND_CLUSTER_NAME
+        value: kms
+      - name: KIND_NETWORK
+        value: kind
+      - group: kubernetes-kms
+
+    steps:
+      - task: GoTool@0
+        inputs:
+          version: 1.16
+
+      - script: make e2e-install-prerequisites
+        displayName: "Install e2e test prerequisites"
+
+      - script: |
+          . scripts/setup-local-registry.sh
+        displayName: "Setup local registry"
+        env:
+          REGISTRY_NAME: $(REGISTRY_NAME)
+          REGISTRY_PORT: $(REGISTRY_PORT)
+
+      - script: |
+          version=$(git tag -l --sort=v:refname | tail -n 1)
+          echo "##vso[task.setvariable variable=LATEST_KMS_VERSION]$version"
+
+          echo "Latest released kms version - $version"
+        displayName: "Get latest released version"
+      
+      - template: manifest-template.yml
+        parameters:
+          registry: mcr.microsoft.com/oss/azure/kms
+          imageName: keyvault
+          imageVersion: $(LATEST_KMS_VERSION)
+
+      - script: |
+          . scripts/setup-kind-cluster.sh &
```

Comment:
This can be replaced with `make e2e-setup-kind`

Reply:
In upgrade we can't use this target since it depends on `setup-local-registry`.  This generates also generates the manifests. For upgrade test we want to keep setup, manifest and cluster creation separate to make sure we install right version of kms